### PR TITLE
vk-messenger: 5.0.1 → 5.2.3, enable on darwin

### PR DESCRIPTION
**Pull request by sikmir**
###### Motivation for this change
Update and enable on darwin.

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @gnidorah

### CODE REVIEW

**Comment by gnidorah** [comment]
Result of `nixpkgs-review pr 102927` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vk-messenger</li>
  </ul>
</details>

File: pkgs/applications/networking/instant-messengers/vk-messenger/default.nix [hunk 1]
```diff
@@ -1,54 +1,74 @@
...
-      sha256 = "1ji23x13lzbkiqfrrwx1pj6gmms0p58cjmjc0y4g16kqhlxl60v6";
+      sha256 = "1c4pxjg9z7wpir0yc0xh6vca0i58jnzylls2v68fplr0mndc37c1";
     };
     x86_64-linux = fetchurl {
       url = "https://desktop.userapi.com/rpm/master/vk-${version}.x86_64.rpm";
-      sha256 = "01vvmia2qrxvrvavk9hkkyvfg4pg15m01grwb28884vy4nqw400y";
+      sha256 = "0619qsyd499iqfzwi3fx9w5ww8lygl9vq1m4ax9kj09ydcm45r2c";
+    };
+    x86_64-darwin = fetchurl {
+      url = "https://desktop.userapi.com/mac/master/vk.dmg";
```
Comment (gnidorah) [inline pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: We must have a fixed version, not "master"

**Review by ghost — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/instant-messengers/vk-messenger/default.nix (hunk 1 above)
Comment (sikmir) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: Upstream doesn't provide versioned link for dmg, unlike rpm.

Comment (gnidorah) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: @sikmir Then on next dmg reupload hash will become invalid. We don't support sources that get overwritten

Comment (sikmir) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: Of course, that's not good, but there is nothing we can do with it. There are a lot of such cases in nixpkgs, for example https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/medio/default.nix#L12

Comment (sikmir) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: And "master" in the url doesn't mean that dmg will be reuploaded on each "commit", they reupload dmg only on version bump.

Comment (gnidorah) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: @sikmir I suppose they just slip away the review. Nix really doesn't support non-fixed schemes.
And actually correct url for medio should be http://dotcolon.net/download/fonts/medio_0200.zip

Comment (sikmir) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: Well, as an alternative way we can use versioned link to web.archive.org, I'm not sure is it OK?

Comment (gnidorah) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:19]: @sikmir That should be ok, see https://github.com/NixOS/nixpkgs/blob/master/pkgs/data/fonts/babelstone-han/default.nix#L8

Comment (sikmir) [reply pkgs/applications/networking/instant-messengers/vk-messenger/default.nix]: Updated.

**Comment by gnidorah** [comment]
Result of `nixpkgs-review pr 102927` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vk-messenger</li>
  </ul>
</details>

**Review by ghost — APPROVED** [review]
Its uses fixed-version url now, though takes some time to download
```fish
$ date; nix-prefetch-url https://web.archive.org/web/20201105174155/https://desktop.userapi.com/mac/master/vk.dmg; date
Чт 05 ноя 2020 22:06:17 MSK
[86.6 MiB DL]
path is '/nix/store/v2a5zxrhmmxg8jsyljgvhlirzwayhkhl-vk.dmg'
08walaii4293qd3hqm2bva7p415bmmqpjfi3svavfx4336wyj2ha
Чт 05 ноя 2020 22:18:22 MSK
```

File: pkgs/applications/networking/instant-messengers/vk-messenger/default.nix [hunk 2]
```diff
@@ -1,54 +1,74 @@
...
+  linux = stdenv.mkDerivation {
+    inherit pname version src meta;
+
+    nativeBuildInputs = [ rpmextract autoPatchelfHook wrapGAppsHook ];
+    buildInputs = (with xorg; [
+      libXdamage libXtst libXScrnSaver libxkbfile
+    ]) ++ [
+      gtk3 nss alsaLib
+    ];
+    runtimeDependencies = [ (lib.getLib udev) libnotify ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/instant-messengers/vk-messenger/default.nix:41]:
```suggestion
    ]) ++ [ gtk3 nss alsaLib ];

    runtimeDependencies = [ (lib.getLib udev) libnotify ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@sikmir please resolve the merge conflict.

**Comment by vitalyavolyn** [comment]
Latest version is 5.2.3 now.

**Pull request merged**